client/engine/store/mockstore.go

```diff
@@ -39,59 +41,90 @@ func (ms MockStore) GetChannelSecretKey() *[]byte {
 func (ms MockStore) GetObjectiveById(id protocols.ObjectiveId) (obj protocols.Objective, ok bool) {
 	// todo: locking
 	obj, ok = ms.objectives[id]
+
+	// return immediately if no such objective exists
+	if !ok {
+		return nil, ok
+	}
+
+	// populate channel data
+	if directfund.IsDirectFundObjective(id) {
```

Comment:
Not very important but instead of using `directfund.IsDirectFundObjective(id)` we could cast `obj` and check the result `dfo, isDirectFund := obj.(*directfund.Objective)` It would save us a line.